### PR TITLE
properly split trainer args and model args in argparser in basic examples

### PR DESCRIPTION
**Pull request by awaelchli**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes the first bullet point in #1937

TODO: check docs if we have more examples of this type
(hyperparameters.rst, weights_loading.rst)

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by awaelchli** [comment]
@Borda this is meant to be a fix PR. ok to leave it for later PR?

File: pl_examples/basic_examples/multi_node_ddp2_demo.py [hunk 1]
```diff
@@ -4,23 +4,19 @@
...
+def main(model_args):
+    """ Main training routine specific for this project """
 
-def main(hparams):
-    """Main training routine specific for this project."""
     # ------------------------
     # 1 INIT LIGHTNING MODEL
     # ------------------------
-    model = LightningTemplateModel(hparams)
+    model = LightningTemplateModel(model_args)
```
Comment (SkafteNicki) [inline pl_examples/basic_examples/multi_node_ddp2_demo.py:19]: Should this not be: `model = LightningTemplateModel(**vars(model_args))` similar to the other scrips?

File: pl_examples/basic_examples/multi_node_ddp_demo.py [hunk 2]
```diff
@@ -4,23 +4,19 @@
...
+def main(model_args):
+    """ Main training routine specific for this project """
 
-def main(hparams):
-    """Main training routine specific for this project."""
     # ------------------------
     # 1 INIT LIGHTNING MODEL
     # ------------------------
-    model = LightningTemplateModel(hparams)
+    model = LightningTemplateModel(model_args)
```
Comment (SkafteNicki) [inline pl_examples/basic_examples/multi_node_ddp_demo.py:19]: Again here: `model = LightningTemplateModel(**vars(model_args))`.

File: pl_examples/basic_examples/multi_node_ddp2_demo.py (hunk 1 above)
Comment (awaelchli) [reply pl_examples/basic_examples/multi_node_ddp2_demo.py:19]: very good, thanks. feel free to make it a suggestion so we can commit it together :)

**Comment by awaelchli** [comment]
this argparser is very confusing.. it simply won't add all arguments to the help command, no matter how I compose the parsers with parents etc..
Unless anybody has an idea, I think I will to close this.

**Comment by Borda** [comment]
> this argparser is very confusing.. it simply won't add all arguments to the help command, no matter how I compose the parsers with parents etc.. Unless anybody has an idea, I think I will to close this.

I think that the way around is adding a class method `from_args` similar to Trainer which would filter arguments based on class init and pass only those allowed...

**Comment by awaelchli** [comment]
I'll close this for now. It seems users anyway prefer things like hydra to separate their configuration in smaller parts. For now it is probably better to keep the examples simple even though the module just takes all the trainer args. 
We can maybe discuss the api suggested by @Borda, but this would be another PR.

**Pull request closed without merging**